### DISCUSSION TITLE
对特定直播间会出现无法获取视频流的情况

### DISCUSSION THREAD

**Discussion by WitherLing** [HMBSbige/BilibiliLiveRecordDownLoader#319] [Q&A] [answered]
原因未知,新添加直播间不存在该类问题,显示错误为"尝试下载直播流时发生错误 没有可用的直播地址"
[BilibiliLiveRecordDownLoader20240326.log](https://github.com/HMBSbige/BilibiliLiveRecordDownLoader/files/14798865/BilibiliLiveRecordDownLoader20240326.log)
[BilibiliLiveRecordDownLoader20240327.log](https://github.com/HMBSbige/BilibiliLiveRecordDownLoader/files/14798866/BilibiliLiveRecordDownLoader20240327.log)

26日仍不存在该问题,27日出现.
![image](https://github.com/HMBSbige/BilibiliLiveRecordDownLoader/assets/74227748/3c2e9f04-83cd-42cc-986b-04d66966794b)

切换版本和使用全新客户端均不能排除问题

**Comment by HMBSbige** [comment]
看上去跟前面几位提问的一样是B站和你网络之间的问题导致的（可能是B站限制）

可以尝试登录或自行调整适合自己的直播间高级设置

**Reply by WitherLing** [reply]
稳定客户端保持了登录状态,全新客户端则没有,二者均出现同样的异常.
高级设置似乎只能调整弹幕服务器的协议,更改后并无帮助.

**Reply by WitherLing** [reply]
感谢提供排障灵感!
猜测应为B站直播间调整,部分直播间无法以TS流录制,flv格式仍可录制.
但在更改默认录制格式时出现问题,似乎无法设置fmp4格式?
设置为fmp4格式时将自动以flv格式录制,即使设置为[fmp4;ts;flv],也会跳过TS格式请求转而直接录制flv格式.

**Reply by HMBSbige** [reply]
> 设置为fmp4格式时将自动以flv格式录制,即使设置为[fmp4;ts;flv],也会跳过TS格式请求转而直接录制flv格式.

请求格式 ≠ B站会返回这种格式

**Comment by Initial-heart-1** [comment]
2周前发现相同问题 录一场下来数千个空ts人都麻了
更改默认格式顺序后故障排除 有人有相同情况的话可能B站侧变动？
或许可以考虑更改程序内置的默认格式顺序以避免可能的问题 @HMBSbige

**Reply by HMBSbige** [reply]
我这到现在也没啥问题，感觉本质不太像是格式问题

**Reply by Initial-heart-1** [reply]
存在复数个相同错误的话那就显然不首先考虑单纯的网络原因了 B站侧总有变动的 只是没分析确实不知道具体发生了什么
只是一个建议 因为这个方案似乎确实解决了目前的问题 并且对于没有问题的用户**看起来**也没什么负面影响
或者再等等看

**Reply by HMBSbige** [reply]
B站各地有不同 cdn，每个 cdn 对不同格式的网络不一样，每个人网络情况又都不一样
所以开放自定义格式就是解决这种问题的
默认格式也没什么问题，flv 终是要被淘汰的，默认 flv 优先是不合理的（我记得 flv 流都没有真原画了）

而且看描述和日志更像是B站服务器ts流更改了什么东西，现在我这没法复现暂时也解决不了

**Reply by Initial-heart-1** [reply]
那考虑给一个更详细日志的dev版本么？我这里目前可以稳定复现该问题，可以运行您指定的版本以提供日志，当前 release 版本的日志看起来似乎没有任何非常有用的信息。

**Reply by HMBSbige** [reply]
解决这种问题日志是看不出的，应该是只能 debug 手动分析

**Reply by HMBSbige** [reply]
不过如果日志是“没有可用的直播地址”的话，可以尝试浏览器能不能打开ts流的直播地址

**Reply by Initial-heart-1** [reply]
如果这样的话 我下次注意一下 再出现的时候试试 debug 看看
因为是考虑 b 站侧的问题我其实考虑分析 b 站侧返回有没有什么异常就可以至少定位问题了 所以首先希望从日志分析
毕竟远端上搞 debug 有些麻烦

**Reply by Initial-heart-1** [reply]
如果这样的话 我下次注意一下 再出现的时候试试 debug 看看
因为是考虑 b 站侧的问题我其实考虑分析 b 站侧返回有没有什么异常就可以至少定位问题了 所以首先希望从日志分析
毕竟远端上搞 debug 有些麻烦

**Reply by HMBSbige** [reply]
其实我在想有没有可能你们遇到的情况是服务器的ts流不支持真原画了😅
看看是不是低画质能录，而高画质不能录

**Comment by HMBSbige** [comment, marked as answer]
本质原因：B站取消了 ts 流的真原画

现在 fmp4 流暂时还有真原画
